### PR TITLE
TP 650 - Schedule workbasket upload to bucket

### PR DESCRIPTION
**Pull request by rachaelcodes**
Triggers workbasket upload on setting workbasket to 'READY_TO_EXPORT'.

### CODE REVIEW

**Comment by rachaelcodes** [comment]
After discussion of the team, have decided to leave the mid-file import to avoid circular import problems for now. We may need to look more into how we queue tasks if we get many more of them.

File: workbaskets/tests/test_models.py [hunk 1]
```diff
@@ -580,28 +674,40 @@ def test_workbasket_accepted_updates_current_tracked_models(new_workbasket, vali
...
-    new_workbasket.export_to_cds()
-    new_footnote.refresh_from_db()
-    assert new_footnote.version_group.current_version.pk == new_footnote.pk
-    new_workbasket.cds_error()
-    new_footnote.refresh_from_db()
-    assert new_footnote.version_group.current_version.pk == original_footnote.pk
+
+    with mock.patch(
+        "exporter.tasks.upload_workbaskets",
+    ) as mock_save:
```
Comment (stuaxo) [inline workbaskets/tests/test_models.py:699]: Patching `exporter.tasks.upload_worbaskets.delay` might be better as `upload_workbaskets` is still our own code, but `delay` is the point where 3rd party code (Celery).

Comment (stuaxo) [reply workbaskets/tests/test_models.py:699]: If you ever want to call the task itself, then thats safe too - (always_eager=True and the broker backend is set to memory://).
The only caveat is saving to s3; some other tests use moto to mock this.

File: workbaskets/models.py [hunk 2]
```diff
@@ -16,6 +16,9 @@
 from common.models import TrackedModel
 from common.models import Transaction
 from common.models.records import TrackedModelQuerySet
+
+# imported lower down to avoid circular import error
+# from exporter.tasks import upload_workbaskets
```
Comment (simonwo) [inline workbaskets/models.py:21]: I don't think you really need this comment, or move the line 20 to replace line 150?

**Review by simonwo — APPROVED** [review]
Great!

File: workbaskets/models.py [hunk 3]
```diff
@@ -144,6 +147,11 @@ def approve(self, user):
             version_group.current_version = obj
             version_group.save()
 
+        # hack to avoid circular import
+        from exporter.tasks import upload_workbaskets
```
Comment (stuaxo) [inline workbaskets/models.py:151]: I had a bit of a look at the importer and its use of `tasks` (after remembering that it also uses celery).
This is fine ... if it points at anything, it's a much bigger refactor so this is probably fine :)

But yeah, there's always importer to stare at :man_shrugging:

**Pull request merged**